.travis.yml

```diff
@@ -1,42 +1,79 @@
 language: node_js
+
 node_js:
-- "8"
+  - "10"
+
+# if using Ubuntu 16 need this library
+# https://github.com/cypress-io/cypress-documentation/pull/1647
+addons:
+  apt:
+    packages:
+    - libgconf-2-4
+
+# cache both yarn and Cypress binary
 cache:
   yarn: true
+  directories:
+    - ~/.cache
+
 git:
   depth: 3
+
 stages:
-- name: npm publish
-  if: branch =~ /^v\d+\.\d+(\.\d+)?(-\S*)?$/ AND tag IS present
-- name: master check and build storybook
+- name: pull_request
+  if: type = pull_request
+
+- name: master
   if: branch = master AND type != pull_request
+
+- name: tag
+  if: branch =~ /^v\d+\.\d+(\.\d+)?(-\S*)?$/ AND tag IS present
+
 before_install:
-- curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.15.2
-- export PATH="$HOME/.yarn/bin:$PATH"
+  - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.15.2
+  - export PATH="$HOME/.yarn/bin:$PATH"
+
 install:
-- yarn install --production=false
-- yarn lint
-- yarn build
+  - yarn install --production=false
+
 jobs:
   include:
-  - stage: npm publish
+  - stage: pull_request
+    script:
+      - yarn lint
+      - yarn test
+      - yarn chromatic
+      - yarn start & npx wait-on http://localhost:9001
+      - yarn cypress run --record
+      - kill $(jobs -p) || true
+    env:
+      - CHROMATIC_APP_CODE="dlpro96xybh"
+      - CYPRESS_RECORD_KEY="3a9347b6-36ab-4a36-823d-709f4078b148"
+
+  - stage: tag
+    script:
+      - yarn build
     deploy:
       provider: script
       script: npm config set //registry.npmjs.org/:_authToken=$NPM_PUBLISH_TOKEN && yarn lerna publish from-package --yes --pre-dist-tag prerelease
       skip_cleanup: true
       on:
         tags: true
-  - stage: master check and build storybook
+
+  - stage: master
     script:
-    - yarn test
-    - yarn build-storybook
+      - yarn chromatic --auto-accept-changes
+      - yarn build-storybook
```

Comment:
I'm unclear if both `yarn chromatic` and `yarn build-storybook` are required. Chromatic builds storybook, but I'm not sure if the directories are the same. I'm leaving both for now.